### ISSUE TITLE
vfu_reset_ctx() should also reset migration state to RUNNING

### ISSUE THREAD

**Issue by changpe1** [nutanix/libvfio-user#458]
1. Start QEMU with migration region
2. Kill QEMU
3. Restart QEMU

I can see the following error messages, maybe libvfio-user should reset the device state to RUNNING too.
```
[2021-04-30 17:52:24.240759] vfio_user.c:1484:vfio_user_log: *ERROR*: /var/run/muser/domain/muser0/8: bad command 2 while device in stopped state
[2021-04-30 17:52:24.240773] vfio_user.c:1476:vfio_user_log: *DEBUG*: /var/run/muser/domain/muser0/8: closing unexpected fd 749 (index 0) from cmd 2
[2021-04-30 17:52:24.240786] vfio_user.c:1484:vfio_user_log: *ERROR*: /var/run/muser/domain/muser0/8: msg0x1: cmd 2 failed: Invalid argument
[2021-04-30 17:52:24.240801] vfio_user.c:1484:vfio_user_log: *ERROR*: /var/run/muser/domain/muser0/8: msg0: bad size 0 in header
[2021-04-30 17:52:24.240812] vfio_user.c:1484:vfio_user_log: *ERROR*: /var/run/muser/domain/muser0/8: msg0: cmd 0 failed: Invalid argument
[2021-04-30 17:52:24.240825] vfio_user.c:1484:vfio_user_log: *ERROR*: /var/run/muser/domain/muser0/8: msg0: not a command req
[2021-04-30 17:52:24.240833] vfio_user.c:1484:vfio_user_log: *ERROR*: /var/run/muser/domain/muser0/8: msg0: cmd 0 failed: Invalid argument
[2021-04-30 17:52:24.240859] vfio_user.c:1484:vfio_user_log: *ERROR*: /var/run/muser/domain/muser0/8: bad command 2 while device in stopped state
[2021-04-30 17:52:24.240867] vfio_user.c:1484:vfio_user_log: *ERROR*: /var/run/muser/domain/muser0/8: msg0x2: cmd 2 failed: Invalid argument
[2021-04-30 17:52:24.240879] vfio_user.c:1484:vfio_user_log: *ERROR*: /var/run/muser/domain/muser0/8: msg0: bad size 0 in header
[2021-04-30 17:52:24.240890] vfio_user.c:1484:vfio_user_log: *ERROR*: /var/run/muser/domain/muser0/8: msg0: cmd 12 failed: Invalid argument
```

**Comment by changpe1** [comment]
link to issue #218

**Comment by tmakatos** [comment]
Yes I think you're right.

**Comment by tmakatos** [comment]
Should be fixed by #457.
